src/main/kotlin/persistence/mapping/UserPreferencesMapper.kt

```diff
@@ -0,0 +1,39 @@
+package persistence.mapping
+
+import data.DayNight
+import data.Language
+import data.UserPreferences
+import data.dao.Dao
+import data.mapping.Mapper
+import persistence.model.IUserPreferencesEntity
+import persistence.model.UserPreferencesEntity
+import javax.inject.Inject
+
+class UserPreferencesMapper @Inject constructor(private val languageRepo: Dao<Language>):
+        Mapper<IUserPreferencesEntity, UserPreferences> {
+
+    override fun mapFromEntity(type: IUserPreferencesEntity): UserPreferences {
+        // gets from database and maps preferred source and target language
+        val preferredSourceLanguage = languageRepo.getById(type.preferredSourceLanguageId).blockingFirst()
+        val preferredTargetLanguage = languageRepo.getById(type.preferredTargetLanguageId).blockingFirst()
+
+        return UserPreferences(
+                id = type.id,
+                preferredTargetLanguage = preferredTargetLanguage,
+                preferredSourceLanguage = preferredSourceLanguage,
+                dayNightMode = enumValues<DayNight>()[type.dayNightMode],
```

Comment:
let's hold off on day/night mode until we plan out implementation of this feature more fully